lms/djangoapps/courseware/access.py

```diff
@@ -562,6 +565,20 @@ def can_load():
         if not group_access_response:
             return group_access_response
 
+        if is_integrity_signature_enabled():
+            # If the user's enrollment mode is non-audit, check if they have an integrity
+            # signature for the course, and deny access if not
+            course_id = str(course_key)
+            enrollment = get_enrollment(user.username, course_id)
```

Comment:
I feel like getting enrollment here is not a good practice. I worry about the performance implication on this call more than the call to waffle flag check. Is there a way to pass in the access check into this function to ensure the same logic?

Reply:
The more I think about it, the more I'm not sure if this logic belongs here for that reason - I'm going to spend some time looking into this.